tests/test_data/packages/small_fake_with_build_deps/setup.py

```diff
@@ -0,0 +1,14 @@
+from __future__ import annotations
+
+from setuptools import setup
+
+setup(
+    name="small_fake_with_build_deps",
+    version=0.1,
+    install_requires=[
+        "small_fake_a",
+    ],
+    extras_require={
+        "b": ["small_fake_b"],
```

Comment:
Could you modify `small_fake_b`'s `pyproject.toml` to also have some build deps come from hooks that would return `small_fake_a`?

Reply:
What are you concerned about?
The new functionality doesn’t affect the resolver.

Reply:
I don't remember exactly... I think I wanted to see that _runtime_ deps of `small_fake_with_build_deps` don't influence _build_ deps of `small_fake_b` since both would depend on the same package but should go into different sets of resulting constraints.

Reply:
Think about it: runtime deps of a build dependency are transitive build deps of a project that uses that build dependency.

Reply:
Do you think this is now covered by `test_all_extras_and_all_build_deps`?